File: base_module/module_pay/src/main/java/com/will/play/pay/WillPay.java
package com.will.play.pay;

import android.app.Activity;

import com.will.play.pay.base.IPayInfo;
import com.will.play.pay.base.IPayStrategy;
import com.will.play.pay.callback.IPayCallback;


public class WillPay {
    public static <T extends IPayInfo> void pay(IPayStrategy<T> payWay, Activity mActivity, T payinfo, IPayCallback callback){
        payWay.pay(mActivity, payinfo, callback);
    }
}

File: base_module/module_pay/src/main/java/com/will/play/pay/WillPay.kt
package com.will.play.pay

import android.app.Activity
import com.will.play.pay.base.IPayInfo
import com.will.play.pay.base.IPayStrategy
import com.will.play.pay.callback.IPayCallback

object WillPay {
    fun <T : IPayInfo> pay(payWay: IPayStrategy<T>, mActivity: Activity, payinfo: T, callback: IPayCallback?) {
        payWay.pay(mActivity, payinfo, callback)
    }
}